src/windowsTerminal.ts

```diff
@@ -166,13 +166,15 @@ export class WindowsTerminal extends Terminal {
     });
   }
 
-  public kill(signal?: string): void {
-    this._deferNoArgs(() => {
-      if (signal) {
-        throw new Error('Signals not supported on windows.');
-      }
-      this._close();
-      this._agent.kill();
+  public kill(signal?: string): Promise<void> {
+    return new Promise((res) => {
+      this._deferNoArgs(() => {
+        if (signal) {
+          throw new Error('Signals not supported on windows.');
+        }
+        this._close();
+        this._agent.kill().then(res);
```

Comment:
Shouldn’t the outer promise be rejected when the `this._agent.kill()` promise fails?

```suggestion
    return new Promise((res, rej) => {
      this._deferNoArgs(() => {
        if (signal) {
          throw new Error('Signals not supported on windows.');
        }
        this._close();
        this._agent.kill().then(res, rej);
```

Reply:
This may or may not be out of scope for this PR, but anyway.

As far as I can tell, the callbacks passed to `this._deferNoArgs` are called later, asynchronously. So it looks like the `throw` here cannot be caught. This is not a new issue in this PR. But with this PR, it’s less expected to throw errors since the function returns a `Promise` – I’d expect the promise to be rejected instead. We have the possibility to do that:

```suggestion
    return new Promise((res, rej) => {
      this._deferNoArgs(() => {
        if (signal) {
          rej(new Error('Signals not supported on windows.'));
        } else {
          this._close();
          this._agent.kill().then(res, rej);
        }
```

However, I don’t know if this is wanted or not by end consumers. They can always check `caughtError.message === 'Signals not supported on windows.'` but 🤷 

If this change is made, this line needs to be updated in node-pty.d.ts:

```
@throws Will throw when signal is used on Windows.
```

Reply:
I think it makes more sense to reject the promise if signal is used, especially if the thrown error would sometimes not be caught. I'm not entirely sure how promise rejections should be described in jsdoc, but changing it to `@throws Will reject when signal is used on Windows.` seems to be fine.

Reply:
Nice!

Note that your commit (unlike my suggestion above) changes behavior: Previously, `this._close()` and `this._agent.kill()` were _not_ run if `signal` is truthy. In your commit, the promise is rejected and then those two methods are run anyway.

Reply:
Whoops, should be fixed now.